src/main/java/ru/svgogin/service/spark/web/SparkController.java

```diff
@@ -37,8 +41,13 @@ public SparkController(SparkService sparkService) {
   }
 
   @PostMapping()
-  public CompanyDto saveCompany(@RequestBody CompanyDto companyDto) {
-    return sparkService.save(companyDto);
+  public ResponseEntity<CompanyDto> saveCompany(@RequestBody CompanyDto companyDto) {
+    var companyInDb = sparkService.findByInn(companyDto.getInn());
+    if (companyInDb.isEmpty()) {
+      return new ResponseEntity<>(sparkService.save(companyDto), HttpStatus.OK);
+    } else {
+      throw new EntityAlreadyExistsException("Company with inn " + companyDto.getInn() + " already exists");
```

Comment:
why do you move this logic from service to controller?

Reply:
Fixed